plenum/server/client_authn.py

```diff
@@ -84,8 +84,13 @@ def getVerkey(self, identifier):
 
 class NaclAuthNr(ClientAuthNr):
 
+    def __init__(self):
+        super().__init__()
+        self.specific_verkey_validation = dict()
```

Comment:
I'm not sure that putting a txn-specific logic into `NaclAuthNr` which is assumed to be quite generic is a good idea. 
A better option could be to move this into `getVerkey` method.

Reply:
BTW please note how it's implemented: 
- The main facade for signature authentication is `ReqAuthenticator`.
- A number of Authenticators (`ClientAuthNr` instances) can be registered there.
- By default Plenum registers `CoreAuthNr` which is responsible for authentication of all write requests
- Indy-Node overrides it and registers `LedgerBasedAuthNr` instead, which is also responsible for authentication of all write requests
- `CoreAuthNr` gets the verkey from a local dict
- `LedgerBasedAuthNr` gets the verkey from IdrCache (assumes that it must be there).

So, we will have to override the method in both implementations.

Reply:
thnx, fixed